src/platforms/evdev/libinput_device.cpp

```diff
@@ -350,7 +365,13 @@ void mie::LibInputDevice::handle_touch_down(libinput_event_touch* touch)
 void mie::LibInputDevice::handle_touch_up(libinput_event_touch* touch)
 {
     MirTouchId const id = libinput_event_touch_get_slot(touch);
-    last_seen_properties[id].action = mir_touch_action_up;
+
+    //Only update last_seen_properties[].action if there is a valid record in the map
+    //for this ID. Otherwise, an invalid "up" event from a bogus panel will
+    //create a fake action.
+    auto const it = last_seen_properties.find(id);
+    if (it != end(last_seen_properties))
+        (*it).second.action = mir_touch_action_up;
```

Comment:
I find this more readable:
```suggestion
        it->second.action = mir_touch_action_up;
```